config/validate.go

```diff
@@ -368,10 +368,17 @@ func (m *Metadata) Validate(data map[string]any) []string {
 					// if the named key is specified then this one must be also
 					otherName := validation.Arg.(string)
```

Comment:
Suggest adding a line that says "this validation cannot be used if the field has a default value".